core/domain/exp_domain.py

```diff
@@ -4353,7 +4353,6 @@ def _fix_drag_and_drop_input_interaction(
                             ele_element = ele_x_at_y_rule['element']
                             assert isinstance(ele_position, int)
                             if ele_position > len(rule_spec_val_x):
-                                invalid_rules.append(rule_spec)
```

Comment:
Why are we removing this? (No info in the PR description for the "why is this needed?")

Reply:
This is also the drag and drop part, I have explained this part in the above comment as why we should not remove the rule. We will not consider the rule as invalid because the user is allowed to place the item at any position available.
Thanks